patches/common/v8/backport_50206308.patch

```diff
@@ -0,0 +1,235 @@
+From 5020630808043615f2f6fbbfba1ea73342bebd69 Mon Sep 17 00:00:00 2001
+From: Daniel Clifford <danno@chromium.org>
+Date: Wed, 28 Mar 2018 13:30:16 +0200
+Subject: [PATCH] Remove legacy C++ implementation of Array#slice
```

Comment:
Just backporting the "Separate species protectors" patch, introduces some changes which were needed in the legacy splice implementation as well. So, instead of making another patch to fix those, it seemed reasonable to backport this commit as well.

Reply:
>  introduces some changes which were needed in the legacy splice implementation as well

But this patch touches slice implementation.

Reply:
Apologies, if it wasn't clear earlier. Legacy slice implementation was removed before the new "Separate species protectors" refactoring that was introduced. Hence, there were some calls to functions which did not exist after the "separate species protector" patch. One solution was what I've done, and the other would have been to make appropriate changes in the legacy slice implementation to use the new functions etc changed after refactoring.

Reply:
Yup its clear now, thanks!

> Hence, there were some calls to functions which did not exist after the "separate species protector" patch

Wouldn't it be just this `isolate->IsSpeciesLookupChainIntact()` call, which can be converted to use the array specific protector. I didn't find any other conflicting call when applying https://chromium-review.googlesource.com/c/v8/v8/+/1030553. The current solution looks good to me, but if the goal is to keep patches minimal I would prefer making that function call change.

Reply:
Yes, that's right. I was also thinking the same, that this is going away with ch67. So, this is going to be short-lived. Thanks!